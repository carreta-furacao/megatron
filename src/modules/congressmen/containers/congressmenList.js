import React, { Component } from 'react'
import { FlatList, Text, StyleSheet, View } from 'react-native'
import PropTypes from 'prop-types'

export class CongressmenList extends Component {
  static propTypes = {
    congressmen: PropTypes.array.isRequired
  }

  renderCongressman = ({ item }) => {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>{item.nome}</Text>
      </View>
    )
  }

  render() {
    return (
      <FlatList
        data={this.props.congressmen}
        keyExtractor={item => item.id}
        renderItem={this.renderCongressman}
      />
    )
  }
}

const styles = StyleSheet.create({
  text: {
    fontSize: 20,
    fontWeight: 'bold'
  },
  container: {
    flex: 1,
    height: 100,
    justifyContent: 'center',
    margin: 100
  }
})
