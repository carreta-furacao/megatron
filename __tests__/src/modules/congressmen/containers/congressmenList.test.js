import React from 'react'
import { shallow } from 'enzyme'

import { CongressmenList } from '../../../../../src/modules/congressmen/containers/congressmenList'
import { congressman } from '../../../../fixtures/congressman.fixture'

const props = {
  congressmen: [congressman]
}

describe('<CongressmenList />', () => {
  it('should have a snapshot', () => {
    const wrapper = shallow(<CongressmenList {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
