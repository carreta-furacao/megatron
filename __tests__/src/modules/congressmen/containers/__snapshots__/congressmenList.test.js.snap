// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CongressmenList /> should have a snapshot 1`] = `
<FlatList
  data={
    Array [
      Object {
        "id": 178957,
        "idLegislatura": 666,
        "nome": "Vinicius Costa",
        "siglaPartido": "FoiceEMartelo",
        "siglaUf": "RS",
        "uri": "https://dadosabertos.camara.leg.br/api/v2/deputados/666",
        "uriPartido": "https://dadosabertos.camara.leg.br/api/v2/partidos/666",
        "urlFoto": "http://www.camara.leg.br/internet/deputado/bandep/666.jpg",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
